src/adapter/graphql/src/queries/search.rs

```diff
@@ -32,13 +33,19 @@ impl Search {
         page: Option<usize>,
         per_page: Option<usize>,
     ) -> Result<SearchResultConnection> {
-        let dataset_registry = from_catalog::<dyn domain::DatasetRegistry>(ctx).unwrap();
-        let dataset_action_authorizer =
-            from_catalog::<dyn domain::auth::DatasetActionAuthorizer>(ctx).unwrap();
+        let (dataset_registry, dataset_action_authorizer) = from_catalog_n!(
+            ctx,
+            dyn domain::DatasetRegistry,
+            dyn domain::auth::DatasetActionAuthorizer
+        );
 
         let page = page.unwrap_or(0);
         let per_page = per_page.unwrap_or(Self::DEFAULT_RESULTS_PER_PAGE);
 
+        // TODO: Private Datasets: PERF: find a way to narrow down the number of records
+        //       to filter, e.g.:
+        //       - Anonymous: get all the public
+        //       - Logged: all available based on ReBAC
```

Comment:
Meaning by having a relation between this account and dataset. "Any relation" is a quick query, it does not need to rewrite Polar as SQL. Polar then would refine the criteria.

Reply:
NB. TODO updated